From 1047d8ff2529d08d8000d3766829a1e1ea0f50a1 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 9 Aug 2018 15:41:59 +0300
Subject: [PATCH 0256/1239] cmd: mvebu: fix partition handling in bubt command

Update the code handling MMC partitions following latest
changes in the basic MMC structures.

Change-Id: I9cd36550cd30726c5382d77ac34cb293aafce325
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
---
 cmd/mvebu/bubt.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/cmd/mvebu/bubt.c b/cmd/mvebu/bubt.c
index ae7515c585..0e15425ed6 100644
--- a/cmd/mvebu/bubt.c
+++ b/cmd/mvebu/bubt.c
@@ -167,8 +167,8 @@ static int mmc_burn_image(size_t image_size)
 	}
 
 #ifdef CONFIG_SYS_MMC_ENV_PART
-	if (mmc->part_num != CONFIG_SYS_MMC_ENV_PART) {
-		err = mmc_switch_part(mmc_dev_num, CONFIG_SYS_MMC_ENV_PART);
+	if (mmc_get_blk_desc(mmc)->hwpart != CONFIG_SYS_MMC_ENV_PART) {
+		err = mmc_switch_part(mmc, CONFIG_SYS_MMC_ENV_PART);
 		if (err) {
 			printf("MMC partition switch failed\n");
 			return err;
@@ -208,8 +208,8 @@ static int mmc_burn_image(size_t image_size)
 	printf("Done!\n");
 
 #ifdef CONFIG_SYS_MMC_ENV_PART
-	if (mmc->part_num != CONFIG_SYS_MMC_ENV_PART)
-		mmc_switch_part(mmc_dev_num, mmc->part_num);
+	if (mmc_get_blk_desc(mmc)->hwpart != CONFIG_SYS_MMC_ENV_PART)
+		mmc_switch_part(mmc, mmc_get_blk_desc(mmc)->hwpart);
 #endif
 
 	return 0;
-- 
2.29.0

